### PR TITLE
Add filter for tracking

### LINKED ISSUE #1899: [Feature Request] Filter by Tracking or untracking
Adding this filter like this will help us to track which manga are tracked and which not 

Capture11111

### PR DESCRIPTION
**Pull request by ghostbear**
closes #1899

Checks if a manga is tracked and checks if you are logged into that tracking service

https://user-images.githubusercontent.com/6576096/104059037-eae9e800-51f4-11eb-96e5-fd5b7e6645b6.mp4

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryPresenter.kt [hunk 1]
```diff
@@ -144,11 +147,23 @@ class LibraryPresenter(
             else !isDownloaded
         }
 
+        val filterFnTracking: (LibraryItem) -> Boolean = tracking@{ item ->
+            if (tracking == State.IGNORE.value) return@tracking true
+
+            val tracks = db.getTracks(item.manga)
+                .executeAsBlocking()
+
+            val isTracking = tracks.map { trackManager.getService(it.sync_id)?.isLogged ?: false }.contains(true)
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryPresenter.kt:156]: Can short circuit with `.any`

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryPresenter.kt [hunk 2]
```diff
@@ -144,11 +147,23 @@ class LibraryPresenter(
             else !isDownloaded
         }
 
+        val filterFnTracking: (LibraryItem) -> Boolean = tracking@{ item ->
+            if (tracking == State.IGNORE.value) return@tracking true
+
+            val tracks = db.getTracks(item.manga)
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryPresenter.kt:153]: Idk how I feel about checking the database for every single manga individually...

Comment (ghostbear) [reply app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryPresenter.kt:153]: Should I instead check the database and add the value to a map that keeps track of database manga id and track id?

**Comment by ghostbear** [comment]
I have changed so it gets all the tracked manga from the database and observers if any changes happen

**Pull request merged**